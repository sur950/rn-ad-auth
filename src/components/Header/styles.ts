import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: 44,
    paddingHorizontal: 20,
    backgroundColor: '#f8f8f8',
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomColor: '#d8d8d8',
    borderBottomWidth: 1,
  },
  closeImg: {
    height: 14,
    width: 14,
  },
  lockImg: {
    height: 12,
    width: 12,
  },
  cancelText: {
    color: '#007aff',
    fontSize: 18,
    fontWeight: '400',
  },
  urlBox: {
    marginLeft: 10,
    flexDirection: 'row',
    alignItems: 'center',
  },
  urlText: {
    fontWeight: 'bold',
    fontSize: 15,
    marginLeft: 5,
  },
});

export default styles;
