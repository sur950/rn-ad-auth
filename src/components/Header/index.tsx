import React from 'react';
import { View, Text, Platform, TouchableOpacity, Image } from 'react-native';
import type { HeaderPropTypes } from '../../types';
import styles from './styles';

const Header = (props: HeaderPropTypes) => {
  const { onCancel, url = 'https://login.microsoftonline.com' } = props;

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={onCancel}>
        {Platform.OS === 'ios' ? (
          <Text style={styles.cancelText}>Cancel</Text>
        ) : (
          <Image
            style={styles.closeImg}
            resizeMode="contain"
            source={require('../../assets/images/close.png')}
          />
        )}
      </TouchableOpacity>
      <View style={styles.urlBox}>
        <Image
          style={styles.lockImg}
          resizeMode="contain"
          source={require('../../assets/images/lock.png')}
        />
        <Text style={styles.urlText}>{url}</Text>
      </View>
    </View>
  );
};

export default Header;
