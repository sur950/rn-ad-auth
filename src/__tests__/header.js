// __tests__ for AuthModal Widget

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Header test', () => {
  it(`should match the Header snapshot`, () => {
    const tree = renderer.create(<Header onCancel={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
