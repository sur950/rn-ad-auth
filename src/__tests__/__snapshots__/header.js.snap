// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should match the Header snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f8f8f8",
      "borderBottomColor": "#d8d8d8",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#007aff",
          "fontSize": 18,
          "fontWeight": "400",
        }
      }
    >
      Cancel
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 10,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
          "marginLeft": 5,
        }
      }
    >
      https://login.microsoftonline.com
    </Text>
  </View>
</View>
`;
